operator/pkg/lbipam/lbipam_fixture_test.go

```diff
@@ -0,0 +1,293 @@
+package lbipam
+
+import (
+	"encoding/json"
+	"reflect"
+	"time"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	cilium_api_v2alpha1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	cilium_fake "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/fake"
+	"github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	core_v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/fake"
+	client_core_v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	k8s_testing "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	poolAUID = types.UID("eb84f074-806e-474e-85c5-001f5780906b")
+	poolBUID = types.UID("610a54cf-e0e5-4dc6-ace9-0e6ca9a4aaae")
+
+	serviceAUID = types.UID("b801e1cf-9e71-455c-9bc8-52c0575c22bd")
+	serviceBUID = types.UID("b415933e-524c-4f83-8493-de2157fc736f")
+	serviceCUID = types.UID("8d820ef0-d640-497a-bc67-b05190bddee6")
+)
+
+var (
+	servicesResource = schema.GroupVersionResource{Group: "", Version: "v1", Resource: "services"}
+	poolResource     = schema.GroupVersionResource{Group: "cilium.io", Version: "v2alpha1", Resource: "ciliumloadbalancerippools"}
```

Comment:
Use `v2alpha1.SchemaGroupVersion.WithResource(...)` etc.. Interestingly the fake client has this whole thing spelled out, but not exported. Weird it isn't part of the normal typed generated client.